### DISCUSSION TITLE
Usage of OpenCL backend instead of PTX for tornadovm-nvidia-openjdk

### DISCUSSION THREAD

**Discussion by andrii0lomakin** [beehive-lab/TornadoVM#500] [Q&A] [answered]
Good day.
May I ask you to provide reasons why only OpenCL backed is used for tornadovm-nvidia-openjdk docker image?
Shouldn't the PTX backend be available, too?

**Comment by stratika** [comment, marked as answer]
hi @andrii0lomakin, as far as I can tell it is just that the dockerfile that is used to build the container does not build TornadoVM has used only the OpenCL argument. We can change that for the next release (end of September). Thanks for pointing it out.

**Reply by andrii0lomakin** [reply]
Thank you @stratika

**Comment by jjfumero** [comment]
Our OpenCL images extend the `nvidia/opencl` base images. 

https://github.com/beehive-lab/docker-tornadovm/blob/master/dockerFiles/Dockerfile.nvidia.jdk21#L3

When we built these images, PTX SDKs were not available for the OpenCL image. Alternatively, you can extend from the `nvidia/cuda` image:

https://hub.docker.com/r/nvidia/cuda/

**Reply by andrii0lomakin** [reply]
Hi @jjfumero 
Yeah, sure, I understand it.

It was mainly an FYI discussion. To point out this behavior, you may want to change.
